portal/src/main/java/com/liferay/faces/portal/component/inputsearch/internal/InputSearchRenderer.java

```diff
@@ -193,34 +214,9 @@ protected void dynamicallyAddChildComponents(InputSearch inputSearch) {
 					// For each client behavior associated with the "action" event:
 					for (ClientBehavior clientBehavior : clientBehaviors) {
 
-						// Determine whether or not the developer added an f:ajax child tag.
-						if (clientBehavior instanceof AjaxBehavior) {
-
-							hasButtonAjaxBehavior = true;
-
-							// Add the InputSearch to the list of components that participate in the "execute" portion
-							// of the JSF partial request lifecycle.
-							AjaxBehavior ajaxBehavior = (AjaxBehavior) clientBehavior;
-							String inputSearchId = inputSearch.getId();
-							Collection<String> execute = new ArrayList<String>();
-							execute.addAll(ajaxBehavior.getExecute());
+						hasButtonAjaxBehavior = true;
 
-							if (execute.contains("@this") || !execute.contains(inputSearchId)) {
-								execute.add(inputSearchId);
-								ajaxBehavior.setExecute(execute);
-							}
-
-							// Add the InputSearch to the list of components that participate in the "render" portion
-							// of the JSF partial request lifecycle.
-							Collection<String> render = new ArrayList<String>();
-							render.addAll(ajaxBehavior.getRender());
-
-							if (render.contains("@this")) {
-								render.remove("@this");
-								render.add(inputSearchId);
-								ajaxBehavior.setRender(render);
-							}
-						}
+						RendererUtil.changeClientBehaviorIds(clientBehavior, inputSearch.getId());
```

Comment:
I moved the changeClientBehaviorIds method back to InputSearchRenderer.java since that is the only class that currently calls it. If we have other components in the future that need to call it, we can move it to RendererUtil.